corehq/apps/users/bulkupload.py

```diff
@@ -544,7 +545,7 @@ def _set_progress(progress):
                         # Without this line, digest auth doesn't work.
                         # With this line, digest auth works.
                         # Other than that, I'm not sure what's going on
-                        user.get_django_user().check_password(password)
+                        user.get_django_user(db_name=get_default_db_aliases()[0]).check_password(password)
```

Comment:
My goal here is to read from the primary DB for the ```User``` model.  This feels non-future-proof because one day we might not store ```User```s in the default DB, but I couldn't think of a better way to determine the primary DB of a model.  Suggestions are welcome.

Reply:
You could use `db_for_read_write(..., write=True)` to be more future-proof.

Alternatively we could create a context manager as Simon mentioned (Perhaps make it override a global setting say `FORCE_PRIMARY`, and use that setting in `MultiDBRouter.db_for_read` to set `write` to True or False)

Reply:
Probably good idea to add a note here on why standbys are avoided.

Reply:
@sravfeyn thanks for the ```db_for_read_write``` pointer